domain/offer/service/offer.go

```diff
@@ -113,3 +113,9 @@ func (s *Service) Offer(
 	err = errors.Errorf("failed to create access for offer %q: %w", args.OfferName, err)
 	return errors.Capture(err)
 }
+
+// ListOffers returns offer details for all offers satisfying the
+// provided filters.
+func (s *Service) ListOffers(ctx context.Context, filters []offer.OfferFilter) ([]*offer.OfferDetails, error) {
```

Comment:
I know people have been doing it elsewhere, but the service layer should have `GetOffers` whatever the API method is called. The lack of consistency at the service layer would tend to suggest to a reader that _getters_ are different to _listers_.

Reply:
We should also be clear in comments at each place the filters are input, that they are or'd together.

Reply:
I'd drop filters on offers like a stone. The offer name is a regex that was run directly on the database in 3.x. We should not be doing that.

https://github.com/juju/juju/blob/0e8973b81315257bcc2e33722efb026ff088376d/cmd/juju/crossmodel/list.go#L173

https://github.com/juju/juju/blob/c6a9c1d9083936c85886b18057dc12c5c24e2f83/state/applicationoffers.go#L763

If we ever need to paginate or filter we should do it holistically.

Reply:
Updated to GetOffers. Header comment addresses how the filters are used.

We can't drop all of the params filters until the API rewrite work, as some of them determine which model to look for offers in: model, model qualifier.

We know this facade and it's api package need an in-depth rewrite already. 

@manadart thoughts on the other filters?

Reply:
Except the command by default is regexp. So **not** providing regexp to the offer name is breaking functionality.

Reply:
The regex `".*%s.*"` is exactly the same as the `LIKE` clause I wrote.

Reply:
Discussed this in person. The offer name should be exact, there is no need for `LIKE`, and this is done in a subsequent patch so unblocking.